cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java

```diff
@@ -191,8 +191,8 @@ private boolean isValidKafkaTopic(String key) {
 
   private boolean isValidPartitionKey(String key) {
     try {
-      LogPartitionType.valueOf(key);
-    } catch (IllegalArgumentException e) {
+      LogPartitionType.valueOf(key.toUpperCase());
+    } catch (IllegalArgumentException | NullPointerException e) {
```

Comment:
E.g. the `isValidKafkaTopic` in the same class should throw NPE if the `cConf.get` returns null (which we do have default in `cdap-default.xml`).

Anyway, no need to block the merging of this PR with this discussion. The code change itself looks good.